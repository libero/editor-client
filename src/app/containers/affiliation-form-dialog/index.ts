export * from './affiliation-form-dialog';
export * from './connected-affiliation-form-dialog';
