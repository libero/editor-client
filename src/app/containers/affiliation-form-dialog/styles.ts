import { makeStyles } from '@material-ui/core/styles';

export const useAffiliationFormStyles = makeStyles((theme) => ({
  root: {},

  inputField: {
    marginBottom: theme.spacing(3)
  },

  buttonPanel: {
    display: 'flex',
    marginTop: theme.spacing(1),
    '& > *:last-child': {
      marginLeft: theme.spacing(2)
    }
  },

  spacer: {
    flexGrow: 1
  }
}));
