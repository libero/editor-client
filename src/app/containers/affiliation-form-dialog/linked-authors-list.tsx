import React, { useState, useCallback } from 'react';
import DeleteIcon from '@material-ui/icons/Delete';
import { get } from 'lodash';
import { FormControl, InputLabel, MenuItem, Select, IconButton } from '@material-ui/core';

import { getAuthorDisplayName, Person } from 'app/models/person';
import { useAffiliationFormStyles } from './styles';
import { ActionButton } from 'app/components/action-button';

interface LinkedAuthorsListProps {
  linkedAuthors: Person[];
  allAuthors: Person[];
  onChange: (selkectedAuthors: Person[]) => void;
}

const renderAuthorSelectListItem = (author: Person) => (
  <MenuItem key={author.id} value={author.id}>
    {getAuthorDisplayName(author)}
  </MenuItem>
);

export const LinkedAuthorsList: React.FC<LinkedAuthorsListProps> = ({ linkedAuthors, allAuthors, onChange }) => {
  const [userLinkedAuthors, setUserLinkedAuthors] = useState<Person[]>(
    linkedAuthors.length > 0 ? linkedAuthors : [null]
  );

  const triggerOnChange = useCallback(
    (authorsList) => {
      onChange(authorsList.filter((author) => Boolean(author)));
    },
    [onChange]
  );

  const getAuthorSelectList = useCallback(
    (selectedAuthor) => {
      const selectedAuthorId = get(selectedAuthor, 'id');
      const linkedIds = new Set(userLinkedAuthors.map((author: Person) => get(author, 'id')));
      return allAuthors.filter(({ id }) => !linkedIds.has(id) || selectedAuthorId === id);
    },
    [userLinkedAuthors, allAuthors]
  );

  const deleteRow = useCallback(
    (index) => () => {
      const updatedList = [...userLinkedAuthors];
      updatedList.splice(index, 1);
      setUserLinkedAuthors(updatedList);
      triggerOnChange(updatedList);
    },
    [userLinkedAuthors, setUserLinkedAuthors, triggerOnChange]
  );

  const updateRow = useCallback(
    (index) => (event) => {
      const updatedList = [...userLinkedAuthors];
      updatedList[index] = allAuthors.find(({ id }) => id === event.target.value);
      setUserLinkedAuthors(updatedList);
      triggerOnChange(updatedList);
    },
    [userLinkedAuthors, allAuthors, triggerOnChange]
  );

  const addEmptyRow = useCallback(() => {
    setUserLinkedAuthors([...userLinkedAuthors, null]);
  }, [userLinkedAuthors, setUserLinkedAuthors]);

  const classes = useAffiliationFormStyles();
  return (
    <div className={classes.inputField}>
      {userLinkedAuthors.map((author: Person, index: number) => (
        <div className={classes.affiliatedAuthorRow} key={author?.id || index}>
          <FormControl key={get(author, 'id', index)} variant="outlined" className={classes.affiliatedAuthorInput}>
            <InputLabel shrink id="author-affiliations-label">
              Affiliated Author
            </InputLabel>
            <Select
              labelId="author-affiliations-label"
              displayEmpty
              value={get(author, 'id', '')}
              onChange={updateRow(index)}
              label="Affiliated Author"
            >
              <MenuItem value={''}>Select affiliated author</MenuItem>
              {getAuthorSelectList(author).map(renderAuthorSelectListItem)}
            </Select>
          </FormControl>
          <IconButton
            classes={{ root: classes.deleteButton }}
            onClick={deleteRow(index)}
            disabled={userLinkedAuthors.length <= 1}
          >
            <DeleteIcon fontSize="small" />
          </IconButton>
        </div>
      ))}
      <ActionButton variant="addEntity" title="Author" onClick={addEmptyRow} />
    </div>
  );
};
