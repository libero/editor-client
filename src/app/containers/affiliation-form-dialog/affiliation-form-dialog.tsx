import React, { SyntheticEvent, useCallback, useState } from 'react';
import { TextField } from '@material-ui/core';
import { useSelector } from 'react-redux';
import { set, cloneDeep } from 'lodash';

import { useAffiliationFormStyles } from './styles';
import { renderConfirmDialog } from 'app/components/prompt-dialog';
import { ActionButton } from 'app/components/action-button';
import { Affiliation, createAffiliation } from 'app/models/affiliation';
import { getAffiliatedAuthors, getAuthors } from 'app/selectors/manuscript.selectors';
import { LinkedAuthorsList } from 'app/containers/affiliation-form-dialog/linked-authors-list';
import { Person } from 'app/models/person';

interface AffiliationFormDialogProps {
  affiliation?: Affiliation;
  allowLinkAuthors?: boolean;
  onAccept: (affiliation: Affiliation, linkedAuthors: Person[]) => void;
  onCancel: () => void;
  onDelete: (affiliation: Affiliation) => void;
}

const labelProps = { shrink: true };

export const AffiliationFormDialog: React.FC<AffiliationFormDialogProps> = (props) => {
  const isNewAffiliation = !props.affiliation;
  const { allowLinkAuthors } = props;

  const affiliatedAuthors = useSelector(getAffiliatedAuthors)(props.affiliation?.id);
  const [userLinkedAuthors, setUserLinkedAuthors] = useState<Person[]>(affiliatedAuthors);
  const allAuthors = useSelector(getAuthors);

  const [affiliation, setAffiliation] = useState<Affiliation>(
    props.affiliation ||
      createAffiliation(undefined, {
        label: '',
        institution: {
          name: ''
        },
        address: {
          city: ''
        },
        country: ''
      })
  );
  const [isConfirmShown, setConfirmShow] = useState<boolean>(false);

  const classes = useAffiliationFormStyles();

  const handleDelete = useCallback(() => {
    setConfirmShow(true);
  }, [setConfirmShow]);

  const handleAcceptDeletion = useCallback(() => {
    setConfirmShow(false);
    props.onDelete(affiliation);
  }, [setConfirmShow, affiliation, props]);

  const handleRejectDeletion = useCallback(() => {
    setConfirmShow(false);
  }, [setConfirmShow]);

  const handleLinkedAuthorsChange = useCallback((authors: Person[]) => {
    setUserLinkedAuthors(authors);
  }, []);

  const handleFormChange = useCallback(
    (event: SyntheticEvent) => {
      const fieldName = event.target['name'];
      const newValue = event.target['value'];
      const newAffiliation = set(cloneDeep(affiliation), fieldName, newValue);
      setAffiliation(newAffiliation);
    },
    [affiliation, setAffiliation]
  );

  const handleDone = useCallback(() => {
    props.onAccept(affiliation, userLinkedAuthors);
  }, [props, affiliation, userLinkedAuthors]);

  return (
    <section className={classes.root}>
      <TextField
        fullWidth
        name="institution.name"
        label="Department/Institute"
        classes={{ root: classes.inputField }}
        InputLabelProps={labelProps}
        variant="outlined"
        multiline
        value={affiliation.institution.name}
        onChange={handleFormChange}
      />
      <TextField
        name="address.city"
        label="City"
        classes={{ root: classes.inputField }}
        InputLabelProps={labelProps}
        variant="outlined"
        value={affiliation.address.city}
        onChange={handleFormChange}
      />
      <TextField
        name="country"
        fullWidth
        label="Country"
        classes={{ root: classes.inputField }}
        InputLabelProps={labelProps}
        variant="outlined"
        value={affiliation.country}
        onChange={handleFormChange}
      />
      {allowLinkAuthors ? (
        <LinkedAuthorsList
          linkedAuthors={userLinkedAuthors}
          allAuthors={allAuthors}
          onChange={handleLinkedAuthorsChange}
        />
      ) : undefined}
      <div className={classes.buttonPanel}>
        {!isNewAffiliation ? (
          <ActionButton variant="outlinedWarning" onClick={handleDelete} title="Delete" />
        ) : undefined}
        <div aria-hidden={true} className={classes.spacer}></div>
        <ActionButton variant="secondaryOutlined" onClick={props.onCancel} title="Cancel" />
        <ActionButton variant="primaryContained" onClick={handleDone} title="Done" />
      </div>
      {isConfirmShown
        ? renderConfirmDialog(
            'You are deleting an affiliation',
            'Deleting an affiliation will remove all associated links with authors. Are you sure want to proceed?',
            handleAcceptDeletion,
            handleRejectDeletion
          )
        : undefined}
    </section>
  );
};
