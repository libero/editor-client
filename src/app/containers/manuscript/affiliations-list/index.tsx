import React, { useCallback } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { IconButton } from '@material-ui/core';
import EditIcon from '@material-ui/icons/Edit';

import { SectionContainer } from 'app/components/section-container';
import { ActionButton } from 'app/components/action-button';
import { getAffiliations } from 'app/selectors/manuscript.selectors';
import { Affiliation } from 'app/models/affiliation';
import { useAffiliationStyles } from './styles';
import * as manuscriptEditorActions from 'app/actions/manuscript-editor.actions';
import { ConnectedAffiliationFormDialog } from 'app/containers/affiliation-form-dialog';

export const AffiliationsList: React.FC<{}> = () => {
  const classes = useAffiliationStyles();
  const affiliations = useSelector(getAffiliations);
  const dispatch = useDispatch();

  const editAffiliation = useCallback(
    (aff: Affiliation) => {
      dispatch(
        manuscriptEditorActions.showModalDialog({
          component: ConnectedAffiliationFormDialog,
          props: { affiliation: aff },
          title: 'Edit Affiliation'
        })
      );
    },
    [dispatch]
  );

  const addAffiliation = useCallback(() => {
    dispatch(
      manuscriptEditorActions.showModalDialog({
        component: ConnectedAffiliationFormDialog,
        props: {},
        title: 'Add Affiliation'
      })
    );
  }, [dispatch]);

  const renderAffiliation = (aff: Affiliation) => (
    <div key={aff.id} className={classes.listItem}>
      <div className={classes.orderLabel}>({aff.label})</div>
      <div className={classes.affiliationInfo}>
        {[aff.institution.name, aff.address.city, aff.country].filter((field) => Boolean(field)).join(', ')}
      </div>
      <IconButton classes={{ root: classes.editButton }} onClick={editAffiliation.bind(null, aff)}>
        <EditIcon fontSize="small" />
      </IconButton>
    </div>
  );

  return (
    <section>
      <SectionContainer label="Affiliations">{affiliations.map(renderAffiliation)}</SectionContainer>
      <ActionButton title="Affiliation" variant="addEntity" onClick={addAffiliation} className={classes.addButton} />
    </section>
  );
};
