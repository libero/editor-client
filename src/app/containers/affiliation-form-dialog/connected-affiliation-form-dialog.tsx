import React, { useCallback } from 'react';
import { Affiliation } from 'app/models/affiliation';
import { useDispatch } from 'react-redux';
import { AffiliationFormDialog } from 'app/containers/affiliation-form-dialog/affiliation-form-dialog';
import { Person } from 'app/models/person';
import * as manuscriptActions from 'app/actions/manuscript.actions';
import { linkAffiliationsAction } from 'app/actions/manuscript.actions';
import * as manuscriptEditorActions from 'app/actions/manuscript-editor.actions';

interface ConnectedAffiliationsFormDialogProps {
  affiliation: Affiliation;
}

export const ConnectedAffiliationsFormDialog: React.FC<ConnectedAffiliationsFormDialogProps> = ({ affiliation }) => {
  const dispatch = useDispatch();
  const isNewAffiliation = !affiliation;

  const closeDialog = useCallback(() => {
    dispatch(manuscriptEditorActions.hideModalDialog());
  }, [dispatch]);

  const handleAccept = useCallback(
    (affiliation: Affiliation, linkedAuthors: Person[]) => {
      if (isNewAffiliation) {
        dispatch(manuscriptActions.addAffiliationAction(affiliation));
      } else {
        dispatch(manuscriptActions.updateAffiliationAction(affiliation));
      }
      dispatch(linkAffiliationsAction({ affiliation, authors: linkedAuthors }));
      closeDialog();
    },
    [closeDialog, dispatch, isNewAffiliation]
  );

  const handleDelete = useCallback(
    (affiliation: Affiliation) => {
      dispatch(manuscriptActions.deleteAffiliationAction(affiliation));
      closeDialog();
    },
    [closeDialog, dispatch]
  );

  return (
    <AffiliationFormDialog
      affiliation={affiliation}
      onAccept={handleAccept}
      onDelete={handleDelete}
      onCancel={closeDialog}
    />
  );
};
