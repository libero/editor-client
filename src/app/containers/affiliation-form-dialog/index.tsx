import React, { SyntheticEvent, useCallback, useState } from 'react';
import { TextField } from '@material-ui/core';
import { useDispatch, useSelector } from 'react-redux';
import { set, cloneDeep } from 'lodash';

import { useAffiliationFormStyles } from './styles';
import * as manuscriptEditorActions from 'app/actions/manuscript-editor.actions';
import * as manuscriptActions from 'app/actions/manuscript.actions';
import { PromptDialog } from 'app/components/prompt-dialog';
import { ActionButton } from 'app/components/action-button';
import { Affiliation, createAffiliation } from 'app/models/affiliation';
import { getAffiliatedAuthors, getAuthors } from 'app/selectors/manuscript.selectors';
import { LinkedAuthorsList } from 'app/containers/affiliation-form-dialog/linked-authors-list';
import { Person } from 'app/models/person';
import { linkAffiliationsAction } from 'app/actions/manuscript.actions';

interface AffiliationFormDialogProps {
  affiliation?: Affiliation;
}

const renderConfirmDialog = (title: string, msg: string, onAccept: () => void, onReject: () => void) => {
  return (
    <PromptDialog
      title={title}
      message={msg}
      isOpen={true}
      onAccept={onAccept}
      onReject={onReject}
      acceptLabel="Delete"
      rejectLabel="Cancel"
      acceptVariant="containedWarning"
      rejectVariant="secondaryOutlined"
    />
  );
};

const labelProps = { shrink: true };

export const AffiliationFormDialog: React.FC<AffiliationFormDialogProps> = (props) => {
  const dispatch = useDispatch();
  const isNewAffiliation = !props.affiliation;

  const affiliatedAuthors = useSelector(getAffiliatedAuthors)(props.affiliation?.id);
  const [userLinkedAuthors, setUserLinkedAuthors] = useState<Person[]>(affiliatedAuthors);
  const allAuthors = useSelector(getAuthors);

  const [affiliation, setAffiliation] = useState<Affiliation>(
    props.affiliation ||
      createAffiliation(undefined, {
        label: '',
        institution: {
          department: '',
          name: ''
        },
        address: {
          city: ''
        },
        country: ''
      })
  );
  const [isConfirmShown, setConfirmShow] = useState<boolean>(false);

  const classes = useAffiliationFormStyles();

  const closeDialog = useCallback(() => {
    dispatch(manuscriptEditorActions.hideModalDialog());
  }, [dispatch]);

  const handleDelete = useCallback(() => {
    setConfirmShow(true);
  }, [setConfirmShow]);

  const handleAccept = useCallback(() => {
    setConfirmShow(false);
    dispatch(manuscriptActions.deleteAffiliationAction(affiliation));
    closeDialog();
  }, [setConfirmShow, affiliation, closeDialog, dispatch]);

  const handleReject = useCallback(() => {
    setConfirmShow(false);
  }, [setConfirmShow]);

  const handleLinkedAuthorsChange = useCallback((authors: Person[]) => {
    setUserLinkedAuthors(authors);
  }, []);

  const handleFormChange = useCallback(
    (event: SyntheticEvent) => {
      const fieldName = event.target['name'];
      const newValue = event.target['value'];
      const newAffiliation = set(cloneDeep(affiliation), fieldName, newValue);
      setAffiliation(newAffiliation);
    },
    [affiliation, setAffiliation]
  );

  const handleDone = useCallback(() => {
    if (isNewAffiliation) {
      dispatch(manuscriptActions.addAffiliationAction(affiliation));
    } else {
      dispatch(manuscriptActions.updateAffiliationAction(affiliation));
    }
    dispatch(linkAffiliationsAction({ affiliation, authors: userLinkedAuthors }));
    closeDialog();
  }, [affiliation, closeDialog, dispatch, isNewAffiliation, userLinkedAuthors]);

  return (
    <section className={classes.root}>
      <TextField
        fullWidth
        name="institution.department"
        label="Department"
        classes={{ root: classes.inputField }}
        InputLabelProps={labelProps}
        variant="outlined"
        value={affiliation.institution.department}
        onChange={handleFormChange}
      />
      <TextField
        fullWidth
        name="institution.name"
        label="Institute"
        classes={{ root: classes.inputField }}
        InputLabelProps={labelProps}
        variant="outlined"
        value={affiliation.institution.name}
        onChange={handleFormChange}
      />
      <TextField
        name="address.city"
        label="City"
        classes={{ root: classes.inputField }}
        InputLabelProps={labelProps}
        variant="outlined"
        value={affiliation.address.city}
        onChange={handleFormChange}
      />
      <TextField
        name="country"
        fullWidth
        label="Country"
        classes={{ root: classes.inputField }}
        InputLabelProps={labelProps}
        variant="outlined"
        value={affiliation.country}
        onChange={handleFormChange}
      />
      <LinkedAuthorsList
        linkedAuthors={userLinkedAuthors}
        allAuthors={allAuthors}
        onChange={handleLinkedAuthorsChange}
      />
      <div className={classes.buttonPanel}>
        {!isNewAffiliation ? (
          <ActionButton variant="outlinedWarning" onClick={handleDelete} title="Delete" />
        ) : undefined}
        <div aria-hidden={true} className={classes.spacer}></div>
        <ActionButton variant="secondaryOutlined" onClick={closeDialog} title="Cancel" />
        <ActionButton variant="primaryContained" onClick={handleDone} title="Done" />
      </div>
      {isConfirmShown
        ? renderConfirmDialog(
            'You are deleting an affiliation',
            'Deleting an affiliation will remove all associated links with authors. Are you sure want to proceed?',
            handleAccept,
            handleReject
          )
        : undefined}
    </section>
  );
};
