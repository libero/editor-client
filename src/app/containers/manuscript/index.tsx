import React from 'react';
import { useSelector } from 'react-redux';
import { Backdrop, CircularProgress } from '@material-ui/core';

import { isManuscriptLoaded } from '../../selectors/manuscript.selectors';
import './styles.scss';
import { ManuscriptToolbar } from './manuscript-toolbar';
import { ManuscriptEditor } from './manuscript-editor';

const renderBackdrop = (): JSX.Element => (
  <Backdrop open={true}>
    <CircularProgress color="inherit" />
  </Backdrop>
);

export const ManuscriptContainer: React.FC = () => {
  const isLoaded = useSelector(isManuscriptLoaded);
  const renderContent = (): JSX.Element => (
    <React.Fragment>
      <ManuscriptToolbar />
      <ManuscriptEditor />
    </React.Fragment>
  );

  return isLoaded ? renderContent() : renderBackdrop();
};
