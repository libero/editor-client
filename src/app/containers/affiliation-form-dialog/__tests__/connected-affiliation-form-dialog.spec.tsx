import React from 'react';
import { create } from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import { mount } from 'enzyme';
import { EditorState } from 'prosemirror-state';

import * as manuscriptActions from 'app/actions/manuscript.actions';
import { getInitialHistory, getLoadableStateSuccess } from 'app/utils/state.utils';
import { ConnectedAffiliationFormDialog } from 'app/containers/affiliation-form-dialog/connected-affiliation-form-dialog';
import { AffiliationFormDialog } from 'app/containers/affiliation-form-dialog/affiliation-form-dialog';

jest.mock('app/containers/affiliation-form-dialog/affiliation-form-dialog', () => ({
  AffiliationFormDialog: () => <div data-cmp="AffiliationsFormDialog"></div>
}));

describe('Author Form Dialog', () => {
  const mockStore = configureMockStore([]);
  let mockState;

  beforeEach(() => {
    mockState = getInitialHistory({
      title: new EditorState(),
      abstract: new EditorState(),
      authors: [],
      keywordGroups: {},
      affiliations: [
        {
          id: 'some_id',
          label: '1',
          institution: {
            name: 'Cambridge University, Boring science'
          },
          address: {
            city: 'Cambridge'
          },
          country: 'UK'
        }
      ]
    });
  });

  it('renders edit affiliation dialog', () => {
    const store = mockStore({
      manuscript: getLoadableStateSuccess(mockState)
    });

    const wrapper = create(
      <Provider store={store}>
        <ConnectedAffiliationFormDialog affiliation={mockState.present.affiliations[0]} />
      </Provider>
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('dispatches update affiliation action', () => {
    const store = mockStore({
      manuscript: getLoadableStateSuccess(mockState)
    });
    jest.spyOn(store, 'dispatch');

    const wrapper = mount(
      <Provider store={store}>
        <ConnectedAffiliationFormDialog affiliation={mockState.present.affiliations[0]} />
      </Provider>
    );

    wrapper.find(AffiliationFormDialog).prop('onAccept')(mockState.present.affiliations[0], []);
    expect(store.dispatch).toHaveBeenCalledWith(
      manuscriptActions.updateAffiliationAction(mockState.present.affiliations[0])
    );

    expect(store.dispatch).toHaveBeenCalledWith(
      manuscriptActions.linkAffiliationsAction({ affiliation: mockState.present.affiliations[0], authors: [] })
    );
  });

  it('dispatches add affiliation action', () => {
    const store = mockStore({
      manuscript: getLoadableStateSuccess(mockState)
    });
    jest.spyOn(store, 'dispatch');

    const wrapper = mount(
      <Provider store={store}>
        <ConnectedAffiliationFormDialog />
      </Provider>
    );

    wrapper.find(AffiliationFormDialog).prop('onAccept')(mockState.present.affiliations[0], []);
    expect(store.dispatch).toHaveBeenCalledWith(
      manuscriptActions.addAffiliationAction(mockState.present.affiliations[0])
    );

    expect(store.dispatch).toHaveBeenCalledWith(
      manuscriptActions.linkAffiliationsAction({ affiliation: mockState.present.affiliations[0], authors: [] })
    );
  });

  it('dispatches delete affiliation action', () => {
    const store = mockStore({
      manuscript: getLoadableStateSuccess(mockState)
    });
    jest.spyOn(store, 'dispatch');

    const wrapper = mount(
      <Provider store={store}>
        <ConnectedAffiliationFormDialog affiliation={mockState.present.affiliations[0]} />
      </Provider>
    );

    wrapper.find(AffiliationFormDialog).prop('onDelete')(mockState.present.affiliations[0]);
    expect(store.dispatch).toHaveBeenCalledWith(
      manuscriptActions.deleteAffiliationAction(mockState.present.affiliations[0])
    );
  });
});
