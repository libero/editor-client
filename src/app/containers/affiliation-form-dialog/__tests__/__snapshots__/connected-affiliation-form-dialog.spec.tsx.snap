// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Author Form Dialog renders edit affiliation dialog 1`] = `
<div
  data-cmp="AffiliationsFormDialog"
/>
`;
